internal/pkg/xray/tree.go

```diff
@@ -62,6 +67,28 @@ func buildTree(traceMap map[string]*Trace, sourceTrace *Trace) (*Tree, error) {
 
 	// Find all the leaf nodes and their complete paths using DFS algorithm
 	var singlePath []*Segment
+
+	linkedTraceIds := maps.Keys(linkedTraceToSegment)
+
+	//get all linked traces in one call
+	if fetchLinkedTraces && len(linkedTraceIds) > 0 && depth < MAX_TREE_DEPTH {
```

Comment:
If the depth exceeds `MAX_DEPTH`, how do we inform customers? Should they know about it?

Reply:
I think informing them somehow is probably a good idea since its possible that they aren't aware of the number of linked traces that should be included. I think the simplest way is to just add another property [here](https://github.com/awslabs/aws-iatk/blob/develop/cmd/iatk/main.go#L98C1-L98C1) with the name `Warnings` and have the BuildTree method return a warning string on successful calls also